src/index.js

```diff
@@ -76,13 +76,20 @@ const translateArrayIntoLink = (linkList, params) => {
     null;
 };
 
-export default (object, linkName, params) => {
+export default (source, linkName, params) => {
   if (!linkName) {
     throw new Error('No link name was passed to hal link resolver');
   }
-  if (object) {
+  if (source instanceof Array) {
+    const links = source.filter(item => item.rel === linkName);
```

Comment:
I think we can afford to use [find](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find) instead of filter.